### ISSUE TITLE
torch.matrix_power: CPU和GPU上的结果不一致

### ISSUE THREAD

**Issue by triumph-wangyuyang** [pytorch/pytorch#89339]
### 🐛 Describe the bug

Test on the CPU:

    import torch
    A = torch.rand([1, 4], dtype=torch.float32)
    n = 3
    torch.matrix_power(A, n)

result: RuntimeError: mat1 and mat2 shapes cannot be multiplied (1x4 and 1x4)

Test on the GPU:

    import torch
    A = torch.rand([1, 4], dtype=torch.float32).cuda()
    n = 3
    torch.matrix_power(A, n)

result: RuntimeError: mat1 dim 1 must match mat2 dim 0

torch.matrix_power throws different exceptions on the CPU and GPU with the same parameters.

### Versions

pytorch: 1.8.0
Python version: 3.8
CUDA/cuDNN version: cuDNN 11.1
GPU models and configuration: RTX3060

**Comment by triumph-wangyuyang** [comment]
The same situation exists for torch.chain_matmul.

Test on the CPU:

    import torch
    a = torch.rand([6, 2], dtype=torch.float64)
    b = torch.rand([2, 2], dtype=torch.float64)
    c = torch.rand([4, 8], dtype=torch.float64)
    d = torch.rand([8, 10], dtype=torch.float64)
    out = torch.chain_matmul(a, b, c, d)

result: RuntimeError: mat1 and mat2 shapes cannot be multiplied (6x2 and 4x10)

Test on the GPU:

    import torch
    a = torch.rand([6, 2], dtype=torch.float64).cuda()
    b = torch.rand([2, 2], dtype=torch.float64).cuda()
    c = torch.rand([4, 8], dtype=torch.float64).cuda()
    d = torch.rand([8, 10], dtype=torch.float64).cuda()
    out = torch.chain_matmul(a, b, c, d)

result: RuntimeError: mat1 dim 1 must match mat2 dim 0

**Comment by lezcano** [comment]
Does this happen in master? If so, please open a new issue and let's discuss it there.

**Comment by triumph-wangyuyang** [comment]
> Does this happen in master? If so, please open a new issue and let's discuss it there.

torch.chain_matmul only throws this exception in the lower version of pytorch1.8. I think it is the same type as the one I posted before, so I mention it here.

**Comment by lezcano** [comment]
Alas, we don't backport fixes. That's why we ask people to send issues testing the relevant code in the lastest version (i.e. nightly or master)

**Comment by triumph-wangyuyang** [comment]
> Alas, we don't backport fixes. That's why we ask people to send issues testing the relevant code in the lastest version (i.e. nightly or master)

Because considering many specific cases, we need to use a lower version of pytorch to conduct experiments, and then find problems during the experiment, so I came here to ask if it can be solved.

**Comment by soumith** [comment]
@triumph-wangyuyang no, we do not backport fixes for older versions. your only option if you want to get past it is to upgrade your use-case to the latest version (i.e. if it was indeed fixed in `master` already)

**Comment by triumph-wangyuyang** [comment]
> @triumph-wangyuyang no, we do not backport fixes for older versions. your only option if you want to get past it is to upgrade your use-case to the latest version (i.e. if it was indeed fixed in already)`master`

understand, thank you
